src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Why/WhyJsonRenderer.cs

```diff
@@ -0,0 +1,116 @@
+                Project = reportModel.ProjectName,
+                Package = reportModel.TargetPackage,
+                DependencyGraphs = dependencyGraphs,
+            };
+
+            var json = JsonSerializer.Serialize(report, new JsonSerializerOptions
+            {
+                WriteIndented = true,
+                IncludeFields = true,
+            });
+
+            _logger.LogMinimal(json);
+        }
+
+        private static List<DependencyGraphContract> ProcessDependencyGraphContracts(Dictionary<string, List<DependencyNode>?> dependencyGraphPerFramework)
+        {
+            List<DependencyGraphContract> dependencyGraphs = [];
+            foreach (var framework in dependencyGraphPerFramework.Keys)
+            {
+                var dependencies = dependencyGraphPerFramework[framework];
+                if (dependencies != null)
+                {
+                    var dependencyGraph = new DependencyGraphContract
+                    {
+                        Framework = framework,
+                        Dependencies = ProcessDependencyNodeContracts(dependencies)
+                    };
+                    dependencyGraphs.Add(dependencyGraph);
+                }
+            }
+            return dependencyGraphs;
+        }
+
+        private static List<DependencyNodeContract> ProcessDependencyNodeContracts(List<DependencyNode> dependencies)
+        {
+            List<DependencyNodeContract> dependencyNodeContracts = [];
+            foreach (var dependency in dependencies)
+            {
+                var dependencyNodeContract = new DependencyNodeContract
+                {
+                    Id = dependency.Id,
+                    Version = dependency.Version,
+                    Dependencies = ProcessDependencyNodeContracts([.. dependency.Children])
+                };
+                dependencyNodeContracts.Add(dependencyNodeContract);
+            }
+            return dependencyNodeContracts;
+        }
+
+        public class WhyJsonReportContract
+        {
+            [JsonPropertyName("version")]
+            public required int Version { get; set; }
+            [JsonPropertyName("parameters")]
+            public required string Parameters { get; set; }
+            [JsonPropertyName("project")]
+            public required string Project { get; set; }
+            [JsonPropertyName("package")]
+            public required string Package { get; set; }
+            [JsonPropertyName("dependencyGraphs")]
+            public required List<DependencyGraphContract> DependencyGraphs { get; set; }
+        }
+
+        public class DependencyGraphContract
+        {
+            [JsonPropertyName("framework")]
+            public required string Framework { get; set; }
+            [JsonPropertyName("dependencies")]
+            public required List<DependencyNodeContract> Dependencies { get; set; }
+        }
+
+        public class DependencyNodeContract
```

Comment:
Can we just use DependencyNode instead?
It seems to contain nearly the same types.

Reply:
I think it's good to keep the class used to render the json separate from the internal model even if they are virtually the same. In the future if someone changes the `DependencyNode` class we don't necessarily want the output of this command to change as well. The json format should be changed deliberately with a bump to the `ReportOutputVersion`.

The other small benefit is that anyone wanting to parse the output json can just copy/paste all of these classes from this one location.

What do you think?

Reply:
> In the future if someone changes the DependencyNode class we don't necessarily want the output of this command to change as well

I was actually thinking the exact opposite :D 
The DependencyNode is part of the model, so I thought we'd rather annotate the model to avoid accidentally forgetting something. 

That being said, we'd always want to support multiple output versions, so maybe the duplicate type is ok. 

@NuGet/nuget-client looking for opinions.